packages/core/src/utils/webhook-destination.ts

```diff
@@ -0,0 +1,28 @@
+export enum WebhookDestinationType {
+  PostUrl = 'post_url',
+}
+
+interface IPostUrlWebhookDestination {
+  url: () => string;
+  type: WebhookDestinationType.PostUrl;
+  getType: () => WebhookDestinationType.PostUrl;
+}
+
+export class PostUrlWebhookDestination implements IPostUrlWebhookDestination {
+  readonly _url: string;
+  constructor(url: string) {
+    this._url = url;
+  }
+
+  public url(): string {
+    return this._url;
+  }
+
+  type: WebhookDestinationType.PostUrl;
+
+  getType(): WebhookDestinationType.PostUrl {
+    return WebhookDestinationType.PostUrl;
+  }
+}
+
+export type WebhookDestination = InstanceType<typeof PostUrlWebhookDestination>;
```

Comment:
can we union in `string` here so that it is valid for them do just do `destination: "https://foo.com/webhook"`?

Reply:
We can, but then we will loose the type safety of the `WebhookDestination.type`

Reply:
hrm. that's a bummer, I do like the idea of being able to `switch` on that, but also since we don't know if/when we are going to even support other destinations, i feel like the 90% use case API for right now is gonna be people just wanting to specify a URL.

Maybe instead of changing this type definition, you could just change the signature of `putWebhook` so that it would accept either a WebhookDestination or a `string`?

Reply:
yea that will work